identity-account/src/events/macros.rs

```diff
@@ -33,46 +33,47 @@ macro_rules! impl_command_builder {
   ($ident:ident { $(@ $requirement:ident $field:ident $ty:ty $(= $value:expr)?),* $(,)* }) => {
     paste::paste! {
       #[derive(Clone, Debug)]
-      pub struct [<$ident Builder>] {
+      pub struct [<$ident Builder>]<'account, K: $crate::identity::IdentityKey> {
+        account: &'account Account,
+        key: K,
         $(
           $field: Option<$ty>,
         )*
       }
 
-      impl [<$ident Builder>] {
+      impl<'account, K: $crate::identity::IdentityKey> [<$ident Builder>]<'account, K> {
         $(
           pub fn $field<VALUE: Into<$ty>>(mut self, value: VALUE) -> Self {
             self.$field = Some(value.into());
             self
           }
         )*
 
-        pub fn new() -> [<$ident Builder>] {
+        pub fn new(account: &'account Account, key: K) -> [<$ident Builder>]<'account, K> {
           [<$ident Builder>] {
+            account,
+            key,
             $(
               $field: None,
             )*
           }
         }
 
-        pub fn finish(self) -> $crate::Result<$crate::events::Command> {
-          Ok($crate::events::Command::$ident {
+        pub async fn apply(self) -> $crate::Result<()> {
+          let account = self.account;
+          let update = $crate::events::Command::$ident {
             $(
               $field: impl_command_builder!(@finish self $requirement $field $ty $(= $value)?),
             )*
-          })
+          };
+          account.apply_command(self.key, update).await?;
+          Ok(())
         }
       }
 
-      impl Default for [<$ident Builder>] {
-        fn default() -> Self {
-          Self::new()
-        }
-      }
-
-      impl $crate::events::Command {
-        pub fn [<$ident:snake>]() -> [<$ident Builder>] {
-          [<$ident Builder>]::new()
+      impl<'account, K: $crate::identity::IdentityKey + Clone> $crate::identity::IdentityUpdater<'account, K> {
+        pub fn [<$ident:snake>](&self) -> [<$ident Builder>]<'account, K> {
+          [<$ident Builder>]::new(self.account(), self.key().clone())
```

Comment:
We could alter the builder to take the `self` instead of `&self` then destruct it to get the `account` and `key` references without cloning. This would prevent re-using an `IdentityUpdater` for more than one command but I'm not convinced multiple commands per updater is necessary given how lightweight creating a new one is.

Reply:
I've got rid of the `Clone` requirement and the `clone()` itself by letting the updater take a reference to the `key` as well. It can still be used as before in the methods example, although I've updated that one.